sdcm/results_analyze/__init__.py

```diff
@@ -146,41 +145,45 @@ def render_to_html(self, results, html_file_path="", template=None):
         return html
 
     def send_email(self, subject, content, html=True, files=()):
-        if self._send_email and self._email_recipients:
+        if self._email_recipients:
             self.log.debug('Send email to {}'.format(self._email_recipients))
             email = Email()
             email.send(subject, content, html=html, recipients=self._email_recipients, files=files)
         else:
-            self.log.warning("Won't send email (send_email: %s, recipients: %s)",
-                             self._send_email, self._email_recipients)
+            self.log.warning("Won't send email (recipients: %s)", self._email_recipients)
 
-    def save_html_to_file(self, results):
+    def save_html_to_file(self, results, file_name, template_file):
         email = BaseEmailReporter()
-        report_file = os.path.join(email.logdir, 'reactor_stall_events_list.html')
+        report_file = os.path.join(email.logdir, file_name)
         self.log.debug('report_file = %s', report_file)
-        template_file = 'results_reactor_stall_events_list.html'
         email.save_html_to_file(results, report_file, template_file=template_file)
         self.log.debug('HTML successfully saved to local file')
         return report_file
 
     def gen_kibana_dashboard_url(self, dashboard_path=""):
         return "%s/%s" % (self._conf.get('kibana_url'), dashboard_path)
 
+    @staticmethod
+    def save_email_data_file(subject, email_data, file_path='email_data.json'):
+        if os.path.exists(file_path):
+            with open(file_path, 'r') as file:
+                data = file.read().strip()
+                file_content = json.loads(data or '{}')
+        else:
+            file_content = dict()
+        file_content[subject] = email_data.copy()
+        with open(file_path, 'w') as file:
+            json.dump(file_content, file)
+
```

Comment:
Need some log debug or info messages about the progress. So if something go wrong, it was easy to find what was a problem

Reply:
added